src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,107 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int MINIMAL_GROWING_STEP = 1;
+    private static final int MINIMAL_SIZE_LIST = 0;
+    private static final int START_INDEX = 0;
+    private static final double MINIMAL_GROWING_RANGE = 1.5;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity(size + MINIMAL_GROWING_STEP);
```

Comment:
Let's better use something like this:
```suggestion
        growIfNeeded();
```